### ISSUE TITLE
listing.json

### ISSUE THREAD

**Issue by mr-each** [EXBO-Studio/stalcraft-database#5]
Hi!

There is a file in each of the "realm" folder with the list of every item in the game. The problem is – it is not. Some items are missing.
IMO, it is the easiest way of running through every single item to get its name and/or ID.
So the question is: will it be updated? 

I'm not very familiar with GitHub API and do not know is there an option to get a tree of a repo, but I found that the only solution to get every possible pair of ITEM_NAME - ITEM_ID is to go through every folder, subfolders and, finally, every item, which is crazy amount of requests especially with GitHub restriction of like 5k per hour. I think this file would be a lifesaver for some kind of an AH monitoring app which relies on item_name more than type-subtype-item_name.

Git clone of a repo would be an option, of course, but you already have like half of the solution =)

**Comment by mihanc** [comment]
I agree with you. Given the structure of the EXBO repository, it seems to me that the data was exported directly from the game without any changes to the structure.
The best way is to write your own parser (or use an existing one), parser the data from the EXBO repository into a format you prefer and save it in your database.

**Comment by NotTheStallion** [comment]
Even if you check all directories and subdirectories, the repo is not up-to-date and some people have submitted to multiple API accesses to be able to check all combinations of APIs. However, it is very slow as the API gives 400 calls per minute and knowing that most IDs have a length of 5 so the complexity of checking all IDs is O(5^36).

I have constructed a dataset on HugginFace (https://huggingface.co/datasets/NotTheStallion/stalcraft-database) that will hold the IDs of all items. Where all API developers can contribute. I have also used a bit of AI to find IDs faster. I will write a report about what i did to explain how i did it.
